src/server/game/Entities/Player/Player.cpp

```diff
@@ -5038,6 +5041,35 @@ void Player::ResurrectPlayer(float restore_percent, bool applySickness)
     }
 }
 
+void Player::SendGhoulResurrectRequest(Player* target)
+{
+    target->m_ghoulResurrectPlayerGUID = GetGUID();
+
+    WorldPacket data(SMSG_RESURRECT_REQUEST, (8 + 4 + 1 + 1 + 1 + 4));
+    data << uint64(GetGUID());
+    data << uint32(0);
+    data << 0;
+    data << uint8(0);
+    data << uint8(0);
+    data << uint32(0);
+    target->GetSession()->SendPacket(&data);
+}
+
+bool Player::IsValidGhoulResurrectRequest(ObjectGuid guid)
```

Comment:
I tested this, and have 2 little problems (not sure how was in retail at that time);
1) You can't talk when you're dead, shows when you try to /s
2) "Release spirit" still appears on screen

As I said, I am not sure about retail (nor I don't have sniffs), but I think that you should be able to talk at least, and the release spirit message not to appear